lib/cn/wbt_builder.c

```diff
@@ -298,9 +298,14 @@ wbt_leaf_publish(struct wbb *wbb)
         assert((void *)entry < sfxp);
 
         if (key_extra) {
+            uint32_t kmd_off;
+
             /* kmdoff is too large for u16 */
             omf_set_lfe_kmd(entry, UINT16_MAX);
-            *(uint32_t *)sfxp = cpu_to_omf32(kin->kmd_off);
+
+            kmd_off = cpu_to_omf32(kin->kmd_off);
+            /* Avoids an alignment issue if setting sfxp directly */
+            memcpy(sfxp, &kmd_off, sizeof(kin->kmd_off));
```

Comment:
Using `sizeof(kin->kmd_off)` makes me nervous because it doesn't make it clear that the OMF spec requires it to be a 4 byte scalar.  See this comment in `lib/cn/omf.h`:
```
/* WBT leaf node entry (v6)
 * Note, if lfe_kmd == U16_MAX, then the actual kmd offset is stored as a LE32
 * value at lfe_koff, and the actual key is stored at lfe_koff + 4.
 */
struct wbt_lfe_omf {
    uint16_t lfe_koff;
    uint16_t lfe_kmd;
} HSE_PACKED;
```
Maybe we should add `#define WBT_LFE_INLINE_KMD_OFF_SIZE 4` to `lib/cn/omf.h` and use that as the length for this `memcpy`.

Reply:
Seems semi reasonable. I'll try to remember to adjust this on my next push.

Reply:
It's as if you used syndney bing to apply my patch...

Reply:
In the pre-existing code, what's up with the "kmdoff is too large for u16" and then calls omf_set-lfe_kmc(entry, UINT16_MAX) ???

Reply:
when decoding, if the field that normally stores the 2-byte kmd offset value is set to UINT16_MAX, then the actual kmd offset is stored in the 4 bytes right before the key.  does that clarify? not entirely sure what @beckerg's question is...

Reply:
cool, that answers it, thx!

Reply:
Reapplied your patch Greg :)